File: Algorithm/Code_Materials/c_or_cpp/charpter17_greedy/level1/find_content_children.cpp
#include <iostream>
#include <algorithm>
#include <vector>

int findContentChildren(std::vector<int> g, std::vector<int> s) {
    std::sort(g.begin(), g.end());
    std::sort(s.begin(), s.end());
    int count = 0;
    int start = s.size() - 1;
    // 遍历孩子的胃口
    for (int index = g.size() - 1; index >= 0; index--) {
        if (start >= 0 && g[index] <= s[start]) {
            start--;
            count++;
        }
    }
    return count;
}

int main() {
    std::vector<int> g = {3, 2, 1, 4, 5};
    std::vector<int> s = {3, 4, 5, 6, 7};
    int count = findContentChildren(g, s);
    std::cout << "Number of content children: " << count << std::endl;
    return 0;
}

File: Algorithm/Leetcode/Code_Materials/Java/src/main/java/com/yugutou/charpter17_greedy/level1/FindContentChildren.java
package com.yugutou.charpter17_greedy.level1;

import java.util.Arrays;

/**
 * LeetCode455 分发饼干
 */
public class FindContentChildren {
    public static void main(String[] args) {
        int[] g = {1, 2, 3};
        int[] s = {1, 1};
        System.out.println(findContentChildren(g,s));
    }

    public static int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int count = 0;
        int start = s.length - 1;
        // 遍历孩子的胃口
        for (int index = g.length - 1; index >= 0; index--) {
            if (start >= 0 && g[index] <= s[start]) {
                start--;
                count++;
            }
        }
        return count;
    }

}
